ripple1d/utils/sqlite_utils.py

```
@@ -131,9 +133,17 @@ def zero_depth_to_sqlite(
     # add control id
     df["reach_id"] = [nwm_id] * len(df)
 
+    df["xs_overtopped"] = check_overtopping(rm, wses)
     insert_data(database_path, table_name, df, plan_suffix, missing_grids, boundary_condition="nd")
 
 
+def check_overtopping(rm: RasManager, wses: pd.DataFrame):
+    """Check if the crossection was overopped."""
+    gdf = rm.current_plan.geom.xs_gdf
+    gdf["overtop_elevation"] = gdf[["left_max_elevation", "right_max_elevation"]].min(axis=1)
+    return wses.gt(gdf["overtop_elevation"], axis=0).any().astype(int)
```

I think this assumes that the two dataframes have the same index.  A more robust approach would be:

wses = wses.merge(gdf, left_index=True, right_index=True, how="left")
return wses["flow_col"] > wses["overtop_elevation"]

---

The dataframes should have the same index. 

The wses dataframe has mutlitple water surface elevations that need to be compared to the overtop_elevation per cross section. The .gt method does this comparison across multiple columns of the wses dataframe (water surface elevation resulting from each profie). The .any() method checks if any of them returned True.

---

They should have the same index, but what if we break something in the future that makes them not have the same index?  Or what if in a RAS update they reverse the order in which they store cross-sections.  Thinks like they could fail silently here giving inaccurate results.  I think merging will be more resilient to those cases.

Also, as I think about it, how="inner" would make it more resilient to mismatched indices.

Again, just a suggestion, but I do think it's better.